packages/headless/src/controllers/facets/range-facet/date-facet/headless-date-filter.ts

```diff
@@ -0,0 +1,207 @@
+import {
+  registerDateFacet,
+  RegisterDateFacetActionCreatorPayload,
+  updateDateFacetValues,
+} from '../../../../features/facets/range-facets/date-facet-set/date-facet-actions';
+import {validateDateFacetOptions} from './headless-date-facet-options';
+
+/**
+ * The options defining a `DateFilter`.
+ */
+export interface DateFilterOptions {
+  /**
+   * The field whose values you want to display in the filter.
+   */
+  field: string;
+
+  /**
+   * A unique identifier for the controller.
+   * By default, a unique random identifier is generated.
+   */
+  facetId?: string;
+
+  /**
+   * Whether to exclude folded result parents when estimating the result count for each facet value.
+   *
+   * @defaultValue `true`
+   */
+  filterFacetCount?: boolean;
+
+  /**
+   * The maximum number of results to scan in the index to ensure that the facet lists all potential facet values.
+   *
+   * Note: A high injectionDepth may negatively impact the facet request performance.
+   *
+   * Minimum: `0`
+   *
+   * @defaultValue `1000`
+   */
+  injectionDepth?: number;
+}
+
+export interface DateFilterRange {
+  /**
+   * The starting value for the date range.
+   */
+  start: string;
+
+  /**
+   * The ending value for the date range.
+   */
+  end: string;
+}
+
+export interface DateFilterProps {
+  /**
+   * The options for the `DateFilter` controller.
+   */
+  options: DateFilterOptions;
+  /**
+   * The initial range that should be applied to the `DateFilter` controller.
+   */
+  initialState?: DateFilterRange;
+}
+
+export interface DateFilterState {
+  /**
+   * The facet ID.
+   * */
+  facetId: string;
+
+  /**
+   * The current selected range.
+   */
+  range?: DateFacetValue;
+
+  /**
+   * `true` if a search is in progress and `false` otherwise.
+   */
+  isLoading: boolean;
+}
+
+/**
+ * The `DateFilter` controller makes it possible to create a date filter.
+ */
+export interface DateFilter extends Controller {
+  /**
+   * Deselects current filter.
+   */
+  deselect(): void;
+
+  /**
+   * Updates the selected range.
+   *
+   * @param range - The date range.
+   */
+  setRange(range: DateFilterRange): void;
+
+  /**
+   * The state of the `DateFacet` controller.
+   */
+  state: DateFilterState;
+}
+
+export function buildDateFilter(
+  engine: SearchEngine,
+  props: DateFilterProps
+): DateFilter {
+  if (!loadDateFilterReducer(engine)) {
+    throw loadReducerError;
+  }
+
+  const controller = buildController(engine);
+  const {dispatch} = engine;
+  const facetId = determineFacetId(engine, props.options);
+  const options: RegisterDateFacetActionCreatorPayload = {
+    ...props.options,
+    currentValues: props.initialState
+      ? [{...props.initialState, endInclusive: true, state: 'selected'}]
+      : [],
+    generateAutomaticRanges: false,
+    facetId,
+  };
+
+  validateDateFacetOptions(engine, options);
+  dispatch(registerDateFacet(options));
+
+  return {
+    ...controller,
+    deselect: () => {
```

Comment:
Is `clear` a better name ?

Reply:
Agreed, just deselect is used in other facets

Reply:
do we want to be consistent with facets? I could easily picture this being used in some implementations for adding filters above the search results or in some kind of modal.